.github/workflows/matlab.yml

```diff
@@ -94,3 +130,48 @@ jobs:
         uses: matlab-actions/run-tests@v1
         with:
           select-by-folder: matlab/test 
+  windows:
+    name: AMD64 Windows 2022 MATLAB
+    runs-on: windows-2022
+    if: ${{ !contains(github.event.pull_request.title, 'WIP') }}
+    steps:
+      - name: Check out repository        
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Install MATLAB
+        uses: matlab-actions/setup-matlab@v1
+      - name: Download Timezone Database
+        shell: bash
+        run: ci/scripts/download_tz_database.sh
+      - name: Install ccache
+        shell: bash
+        run: ci/scripts/install_ccache.sh 4.6.3 /usr
+      - name: Setup ccache
+        shell: bash
+        run: ci/scripts/ccache_setup.sh
+      - name: Get Date
+        id: get-date
+        shell: bash
+        run: |
+          echo "date=$(/bin/date -u "+%Y%m%d")" >> $GITHUB_OUTPUT
+      - name: Cache ccache
+        uses: actions/cache@v3
+        with:
+          path: |
+            cpp/**
+          key: cpp-ccache-for-matlab-windows-${{ steps.get-date.outputs.date }}-${{ hashFiles('cpp/**') }}
```

Comment:
If you hash the cpp files why also add the date? Why is a cache from yesterday for state x not valid today?

Reply:
Hi Jacob, that is a good question. I added the date such that we could track the history about when the latest cache was generated. The restore-key doesn't match the date. So it should find the latest cache even though the creation date in the key name is not the current date. My current version doesn't reduce build time. I need to figure out why, either the changing-date causing the problem or the value of "path" is not right. If you know the right value for files under cpp/**, please teach me. I really appreciate it. In the meantime, I'll do exercise by trying different values. Sorry for so many trial submissions.

Reply:
Ah I see, makes sense :)

It looks like it is working now: https://github.com/mathworks/arrow/actions/runs/5204445685/jobs/9388659172#step:9:73 :tada: